src/cluster/placement/storage/helper.go

```diff
@@ -165,23 +165,13 @@ func (h *stagedPlacementHelper) PlacementProto() (proto.Message, int, error) {
 	return ps, value.Version(), err
 }
 
-// GenerateProto generates a proto message with the placement appended to the snapshots.
+// GenerateProto generates a proto message with placement slice
+// containing only single active placement - the specified placement with cutover time set to 0.
+// This ensures backward comapatiblity with clients that rely on staged placement
+// and expect to find at least one placement snapshot having CutoverNanos < now.
 func (h *stagedPlacementHelper) GenerateProto(p placement.Placement) (proto.Message, error) {
-	ps, _, err := h.placements()
-	if err != nil && err != kv.ErrNotFound {
-		return nil, err
-	}
-
-	if l := len(ps); l > 0 {
-		lastCutoverNanos := ps[l-1].CutoverNanos()
-		// When there is valid placement in the snapshots, the new placement must be scheduled after last placement.
-		if lastCutoverNanos >= p.CutoverNanos() {
-			return nil, fmt.Errorf("invalid placement: cutover nanos %d must be later than last placement cutover nanos %d",
-				p.CutoverNanos(), lastCutoverNanos)
-		}
-	}
-
-	ps = append(ps, p)
+	active := p.SetCutoverNanos(0)
```

Comment:
Ah this is smart, somehow when I envisioned this change years ago it's supposed to be way more involving than what this diff looks like. 

My memory on this is a bit blurry now, can you remind me what does the traffic look like for this case:
p0: s0 goes to i1 and i2
p1: s0 cutoff from i1 and i2 and cutover to i3 and i4 on t1

Since clients out there could be looking at p0 or p1, how are we guaranteeing the correctness of traffic?

Reply:
Aggregator clients [use stagedPlacementWatcher](https://github.com/m3db/m3/blob/master/src/cluster/placement/staged_placement_watcher.go#L137) which picks the [latest placement](https://github.com/m3db/m3/blob/9981a01f23eb668aa392bb654b9d5956d024d6de/src/cluster/placement/staged_placement.go#L171) that has cutover time <= now.

However, the client itself determines whether a shard is writeable based its `now` time and the shard's cutover/cutoff time: https://github.com/m3db/m3/blob/master/src/aggregator/client/tcp_client.go#L308

So, the latest placement must reflect "correct" state of all the instances and shards which is the case in practice.